src/libraries/System.Text.Json/tests/Common/UnsupportedTypesTests.cs

```diff
@@ -124,6 +124,20 @@ async Task RunTest<T>(T value)
                     serialized = await Serializer.SerializeWrapper(obj, new JsonSerializerOptions { IgnoreNullValues = true });
                     Assert.Equal(@"{}", serialized);
                 }
+
+#if !BUILDING_SOURCE_GENERATOR_TESTS
```

Comment:
Why isn't this being run in source-gen mode?

Reply:
Polymorphic scenaria will throw `InvalidOperationException` before we get the chance to exercise the changed codepath.

Reply:
Got it. FWIW when there are consequential divergences in source-gen/reflection behavior, we add a separate test variant for source-gen, e.g. [here](https://github.com/dotnet/runtime/blob/424a09cb81c678fb1ba1c27211b80aba2de070ad/src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/Serialization/PropertyVisibilityTests.cs#L28).